From 536c61daa4eb0bd3f811153c61a1ef81a6d68c3e Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Thu, 9 Apr 2020 08:54:02 +0800
Subject: [PATCH] mmc: dw_mmc-rockchip: Enable xfer timer for all Rockchip SoCs

xfer timer was introduced to solve native defects for RK3308
platform but it seems some others also suffer from it. Enable
it would not put any burden on the normal driver process but
just ensure it has a dead-line weapon to fight for any issues
preventing the IP to work normally.

Change-Id: I0b6914d46788f4df99a7d49c608c3336976f78c6
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/mmc/host/dw_mmc-rockchip.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/mmc/host/dw_mmc-rockchip.c b/drivers/mmc/host/dw_mmc-rockchip.c
index 7ce7a3cc95a8..37d043baa920 100644
--- a/drivers/mmc/host/dw_mmc-rockchip.c
+++ b/drivers/mmc/host/dw_mmc-rockchip.c
@@ -301,12 +301,7 @@ static int dw_mci_rockchip_init(struct dw_mci *host)
 				    "rockchip,rk3288-dw-mshc"))
 		host->bus_hz /= RK3288_CLKGEN_DIV;
 
-	if (of_device_is_compatible(host->dev->of_node,
-				    "rockchip,rk3308-dw-mshc"))
-		host->need_xfer_timer = true;
-	else
-		host->need_xfer_timer = false;
-
+	host->need_xfer_timer = true;
 	return 0;
 }
 
-- 
2.35.3

